scripts/maint/pre-push.git-hook

```diff
@@ -52,6 +43,19 @@ do
 			range="$remote_sha..$local_sha"
 		fi
 
+                if ref_is_upstream_branch "$local_ref" == 0 ||
```

Comment:
Should we be checking local_ref at all, or only remote_ref ?

Reply:
I suppose we should check both, to cover weird situations of local regular branch being pushed to remote upstream branch, and vice versa?

Reply:
In that case, I think we should explicitly check for match; pushing maint-0.3.5 to maint-0.3.4 is as bad as pushing bug12345 to maint-0.3.4.

Reply:
a94c3328829482e0da7a175e3dee089b9113d553 and 59c26154e80d589a069e54a9aa52eca2e14bf322 fixes that.